scripts/compile-svg-sprite.js

```diff
@@ -0,0 +1,115 @@
+const fs = require('fs');
+const path = require('path');
+const jsdom = require('jsdom').JSDOM;
+
+// Grab the passed-in filenames from the command
+// line. These should have been provided from a glob
+// pattern.
+const inputPaths = process.argv.slice(2);
+
+// The directory where the output svg should
+// be placed when compiled
+const outputPath = path.resolve(
+  __dirname,
+  "..",
+  "web/themes/new_weather_theme/assets/images/",
+  "spritesheet.svg"
+);
+
+// We use a global JSDOM instance for
+// easily manipulating the elements as
+// we create them. This also parses the XML
+// easily into HTML friendly SVG.
+const dom = new jsdom(`<!DOCTYPE html><html><head></head>/><body></body>/></html>`);
+const window = dom.window;
+const document = window.document;
+
+const getBasename = (filepath) => {
+  return path.basename(filepath, ".svg");
+};
+
+/**
+ * Given a path filename, make the
+ * basename into an id-friendly string
+ *
+ * @param str string | path The filename or path
+ * @return string An id-friendly string
+ */
+const filenameToId = (str) => {
+  const basename = getBasename(str).toLowerCase();
+  return basename.replace(/\s/g, "-");
+};
+
+/**
+ * Removes extra lines from string
+ *
+ * @param str string The source string
+ * @return string A string with extraneous
+ * newlines removed
+ */
+const cleanupWhitespace = (str) => {
+  const lines = str.split("\n");
+  return lines.map(line => {
+    return line.replace(/^\w+\n$/, "");
+  }).filter(line => {
+    return line !== "";
+  }).join("\n");
+};
+
+/**
+ * Removes the XML header from the string
+ *
+ * @param str string The source xml string
+ * @return string The cleaned output xml string
+ */
+const removeXMLDeclaration = str => {
+  const lines = str.split("\n");
+  return lines.filter(line => {
+    return !line.startsWith("<?xml");
+  }).join("\n");
```

Comment:
I'm partial to regular expressions, but I'll defer to you on readability. I think what you have here is probably more readable, tbh.

```suggestion
 return str.replace(/\n<?xml[^\n]*/gm, "");
```